File: clients/asset-client/src/constants/ItemPermissionConstant.ts
import { createConstant } from "./Constant";

const ItemPermissionConstant = createConstant(
  {
    READ: "READ",
    WRITE: "WRITE",
    DELETE: "DELETE",
    DOWNLOAD: "DOWNLOAD",
    FOLDER: "FOLDER",
    FILE: "FILE",
    ICON_FOLDER: "ICON_FOLDER",
    ICON_FILE: "ICON_FILE",
    ICON_PDF: "ICON_PDF",
    ICON_WORD: "ICON_WORD",
    ICON_EXCEL: "ICON_EXCEL",
    ICON_POWERPOINT: "ICON_POWERPOINT",
    ICON_IMAGE: "ICON_IMAGE",
    ICON_VIDEO: "ICON_VIDEO",
    ICON_AUDIO: "ICON_AUDIO",
    ICON_ARCHIVE: "ICON_ARCHIVE",
    ICON_TEXT: "ICON_TEXT",
  } as const,
  {
    READ: { displayName: "Quyền xem" },
    WRITE: { displayName: "Quyền sửa" },
    DELETE: { displayName: "Quyền xóa" },
    DOWNLOAD: { displayName: "Quyền tải về" },
    FOLDER: { displayName: "Thư mục" },
    FILE: { displayName: "File" },
    ICON_FOLDER: { displayName: "" },
    ICON_FILE: { displayName: "" },
    ICON_PDF: { displayName: "" },
    ICON_WORD: { displayName: "" },
    ICON_EXCEL: { displayName: "" },
    ICON_POWERPOINT: { displayName: "" },
    ICON_IMAGE: { displayName: "" },
    ICON_VIDEO: { displayName: "" },
    ICON_AUDIO: { displayName: "" },
    ICON_ARCHIVE: { displayName: "" },
    ICON_TEXT: { displayName: "" },
  }
);

export default ItemPermissionConstant;

File: API/Hinet.Service/Constant/ItemPermissionConstant.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Hinet.Service.Constant
{
    public class ItemPermissionConstant
    {
        // Permission Types
        [DisplayName("Quyền xem")]
        public static string READ => "READ";

        [DisplayName("Quyền sửa")]
        public static string WRITE => "WRITE";

        [DisplayName("Quyền xóa")]
        public static string DELETE => "DELETE";

        [DisplayName("Quyền tải về")]
        public static string DOWNLOAD => "DOWNLOAD";

        [DisplayName("Thư mục")]
        public static string FOLDER => "FOLDER";

        [DisplayName("File")]
        public static string FILE => "FILE";

        public static string ICON_FOLDER => "FolderIcon";
        public static string ICON_FILE => "FileIcon";
        public static string ICON_PDF => "PdfIcon";
        public static string ICON_WORD => "WordIcon";
        public static string ICON_EXCEL => "ExcelIcon";
        public static string ICON_POWERPOINT => "PowerPointIcon";
        public static string ICON_IMAGE => "ImageIcon";
        public static string ICON_VIDEO => "VideoIcon";
        public static string ICON_AUDIO => "AudioIcon";
        public static string ICON_ARCHIVE => "ArchiveIcon";
        public static string ICON_TEXT => "TextIcon";
    }
}
